### PR TITLE
remove current award year constant, use proper year for URN and award_year field

### PR DESCRIPTION
**Pull request by dreamfall**

### CODE REVIEW

File: app/models/form_answer.rb [hunk 1]
```diff
@@ -61,6 +59,7 @@ class FormAnswer < ActiveRecord::Base
 
   begin :callbacks
     before_create :set_account
+    before_save :set_award_year, unless: :award_year?
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: spec/models/form_answer_spec.rb [hunk 2]
```diff
@@ -30,80 +30,81 @@
...
     let(:form_answer) { FactoryGirl.create(:form_answer, :promotion) }
 
     before do
       allow_any_instance_of(FormAnswer).to receive(:eligible?) { true }
     end
 
-    it 'creates supporters for ep form' do
-      form_answer.document = { supporters: [{ email: 'supporter1@example.com' }.to_json, { email: 'supporter2@example.com' }.to_json] }
+    it "creates supporters for ep form" do
+      form_answer.document = { supporters: [{ email: "supporter1@example.com" }.to_json, { email: "supporter2@example.com" }.to_json] }
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Line is too long. [135/100]

File: spec/models/form_answer_spec.rb [hunk 3]
```diff
@@ -30,80 +30,81 @@
...
+        expect(form_answer.supporters.first.email).to eq("supporter1@example.com")
       end
     end
   end
 
-  describe '#company_or_nominee_from_document' do
+  describe "#company_or_nominee_from_document" do
     subject{ build(:form_answer, kind, document: doc)}
-    let(:c_name){ 'company name'}
+    let(:c_name){ "company name"}
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Space missing to the left of {.<br>Space missing inside }.

File: spec/models/form_answer_spec.rb [hunk 4]
```diff
@@ -30,80 +30,81 @@
...
 
-    context 'promotion form' do
-      let(:doc){ { 'organization_name' => " #{c_name} "}}
+    context "promotion form" do
+      let(:doc){ { "organization_name" => " #{c_name} "}}
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Space missing to the left of {.<br>Space missing inside }.<br>Space inside } missing.

File: spec/models/form_answer_spec.rb [hunk 5]
```diff
@@ -30,80 +30,81 @@
...
-      it 'gets the orgzanization name first' do
+      it "gets the orgzanization name first" do
         expect(subject.company_or_nominee_from_document).to eq(c_name)
       end
 
-      context 'organization name blank' do
-        let(:doc){ {'nominee_first_name' => c_name}}
-        it 'gets the nominee name' do
+      context "organization name blank" do
+        let(:doc){ {"nominee_first_name" => c_name}}
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Space missing to the left of {.<br>Space missing inside }.<br>Space inside { missing.<br>Space inside } missing.

File: spec/models/form_answer_spec.rb [hunk 6]
```diff
@@ -30,80 +30,81 @@
...
+        it "gets the nominee name" do
           expect(subject.company_or_nominee_from_document).to eq(c_name)
         end
       end
     end
 
-    context 'innovation form' do
-      let(:doc){ {'company_name' => c_name } }
+    context "innovation form" do
+      let(:doc){ {"company_name" => c_name } }
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Space missing to the left of {.<br>Space inside { missing.

File: app/models/account.rb [hunk 7]
```diff
@@ -28,7 +28,7 @@ def collaborators_without(user)
   end
 
   def has_trade_award_in_this_year?
-    form_answers.for_year("20#{FormAnswer::CURRENT_AWARD_YEAR}".to_i).
+    form_answers.for_year(Date.today.year + 1).
```
Comment (RuslanHamidullin) [inline app/models/account.rb]: hmm
```
Date.today.year + 1
```
will be => 16. 
Just to confirm: Users can apply on award only for next year?

**Comment by RuslanHamidullin** [comment]
:+1: Good to merge ^ some hudon warnings

File: spec/models/form_answer_spec.rb [hunk 8]
```diff
@@ -16,94 +16,100 @@
...
+  it "sets account on creating" do
     form_answer = FactoryGirl.create(:form_answer)
     expect(form_answer.account).to eq(form_answer.user.account)
   end
 
-  context 'URN' do
+  context "URN" do
     before do
-      ["urn_seq_trade","urn_seq_innovation","urn_seq_development","urn_seq_promotion"].each { |seq|
+      ["urn_seq_trade","urn_seq_innovation","urn_seq_development","urn_seq_promotion"].each do |seq|
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Space missing after comma.

File: spec/models/form_answer_spec.rb [hunk 9]
```diff
@@ -16,94 +16,100 @@
...
+      let(:kind) { :promotion }
+      it "gets the orgzanization name first" do
         expect(subject.company_or_nominee_from_document).to eq(c_name)
       end
 
-      context 'organization name blank' do
-        let(:doc){ {'nominee_first_name' => c_name}}
-        it 'gets the nominee name' do
+      context "organization name blank" do
+        let(:doc) { {"nominee_first_name" => c_name} }
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Space inside { missing.<br>Space inside } missing.

File: spec/models/form_answer_spec.rb [hunk 10]
```diff
@@ -16,94 +16,100 @@
...
         end
       end
     end
 
-    context 'innovation form' do
-      let(:doc){ {'company_name' => c_name } }
-      let(:kind){ :innovation }
-      it 'gets the company name' do
+    context "innovation form" do
+      let(:doc) { {"company_name" => c_name } }
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Space inside { missing.

File: spec/models/form_answer_spec.rb [hunk 11]
```diff
@@ -11,99 +11,105 @@
     end
 
     it "award_type" do
-      should validate_inclusion_of(:award_type).
-             in_array(FormAnswer::POSSIBLE_AWARDS)
+      should validate_inclusion_of(:award_type)
+        .in_array(FormAnswer::POSSIBLE_AWARDS)
```
Comment (matthewford) [inline spec/models/form_answer_spec.rb]: Place the . on the previous line, together with the method call receiver.

**Pull request merged**